llama.cpp

```diff
@@ -6701,6 +6707,10 @@ static int llama_decode_internal(
         n_threads = 1;
     }
 
+#ifdef GGML_USE_SYCL
+    n_threads = 1;
+#endif
+
```

Comment:
This needs to be removed.

Reply:
set `n_threads=1` will increase the performance.
The original code set it in above code block by call `ggml_cpu_has_sycl()` defined in internal ggml lib.
In MacOS CI, the script use the external ggml lib, instead of internal ggml lib.

So, original code will lead to compile fault in MacOS.

This code is a workaround to fix the CI fault of MacOS case:

Reply:
This was added to fix CI in macos build  since architectures does not support sycl.

Reply:
After https://github.com/ggerganov/llama.cpp/pull/4766, the CPU threads are no longer spinning while the other backends are running (there is an exception for prompt processing when not fully offloading the model, but that will also be changed in the near future). Changing the number of threads will not improve the performance of the SYCL backend with full offload, however it will negatively affect performance when using partial offloading, since the CPU portions of the model will only use one thread.

You can ignore the CI failures due to outdated ggml in the MacOS tests.

Reply:
reverted in https://github.com/ggerganov/llama.cpp/pull/2690/commits/05b7f9be6f24104071344f1ff8c73273c84430c9